src/xenia/gpu/texture_dump.cc

```diff
@@ -59,17 +59,17 @@ void TextureDump(const TextureInfo& src, void* buffer, size_t length) {
   switch (src.format) {
     case xenos::TextureFormat::k_DXT1: {
       dds_header.pixel_format.flags = 0x4u;
-      dds_header.pixel_format.fourcc = '1TXD';
+      dds_header.pixel_format.fourcc = multi_char('1', 'T', 'X', 'D');
```

Comment:
I think it would be more convenient to have little-endian and big-endian versions of `multi_char`. This is supposed to mean "DXT1", not "1TXD".

Reply:
mh. we have be template to convert to host endian. Likewise bit shifting always happens in host endian.
how exactly would these endian versions come into play?

Reply:
There probably needs to be a function for generating a FourCC ('A', 'B', 'C', 'D') that would match bytes `ABCD` if stored to a file as uint32_t on the host. With guest file structures it may be a bit confusing though.

Reply:
I think that is what the multi_char function does right now, isnt it?

Reply:
I added a new struct `fourcc_t`
It has methods to retrieve `host`, `be` and `le` variants (`host==le` on x64 of course)
Magic constant now are of that type. Some constants are used in different endians so that makes it easier.
Internally, fourccs are stored as `be`. This way compiler should hopefully be able to short circuit assignments/comparisons with `be<uint32_t>` values.

Reply:
The idea was to internally accept the characters in the natural order (DXT1 rather than 1TXD), but to be able to do something like
```
DDSHeader dds_header;
dds_header.four_cc = make_fourcc_le('D', 'X', 'T', '1');
```
where `make_fourcc_le` would be doing `'D'|('X'<<8)|('T'<<16)|('1'<<24)` on a little-endian host and `'1'|('T'<<8)|('X'<<24)|('D'<<24)` on a big-endian one, and then `fwrite` the whole `dds_header`. Using a separate FourCC would make such `fwrite` impossible if `fourcc_t` is in `DDSHeader`, or require an unneeded trip through the big-endian representation if it's `uint32_t`.